### PR TITLE
IN-610-[SDK]-Scans enhance test coverage

### PR DESCRIPTION
**Pull request by bhavin-te**
# Description

This PR contains code-changes for IN-610 - Enhanced Test Coverage for Scans[SDK].

Module : `tenable\io\scans.py `
Statements covered by Test-cases (%) : 93%

Added test cases to cover maximum statement of Scans module. Earlier test cases covered only 87% statement of scans module. Now 93% statement is covered.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

To test code-changes of this PR, I have run testcases of `pyTenable\tests\io\test_scans.py`. All testcases of this file working fine and getting passed. please find attached snapshot.

TC Scans

Also I have run all test-cases of `pyTenable\tests `and it is getting passed. please find attached snapshot.

TC Scans All

**Test Configuration**:
* Python Version(s) Tested:
* Tenable.sc version (if necessary):

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by MitaliUpadhyay — APPROVED** [review]

**Review by alpesh-te — APPROVED** [review]
LGTM!

File: tests/io/test_scans.py [hunk 1]
```diff
@@ -2059,3 +2059,42 @@ def test_scan_create_scan_document_credentials_pass(api):
...
         getattr(api.scans, '_create_scan_document')(
             {'credentials': credential})
+
+
+@pytest.mark.vcr()
+def test_block_while_running(api):
+    '''
+    test _block_while_running with expected type.
+    '''
+    api.scans._block_while_running(scan_id=583, sleeper=5)
```
Comment (aseemsavio) [inline tests/io/test_scans.py:2069]: Is an assertion not necessary here?

File: tests/io/test_scans.py [hunk 2]
```diff
@@ -2059,3 +2059,42 @@ def test_scan_create_scan_document_credentials_pass(api):
...
+    assert "Scan not found" in not_found_error.value.msg, \
+        "Invalid type validation error for scan_id parameter is not raised by test-case."
+
+
+@pytest.mark.vcr()
+def test_scan_history_sort_direction(api):
+    '''
+    test scan history for sort direction.
+    '''
+    api.scans.history(scan_id=11, limit=10, offset="1", pages=3, sort=(('name', 'asc'), ('description', 'desc')))
```
Comment (aseemsavio) [inline tests/io/test_scans.py:2088]: The `history()` function returns a `ScanHistoryIterator`. Is there a reason for skipping the assertion?

**Review by varap11 — APPROVED** [review]

File: tests/io/test_scans.py (hunk 1 above)
Comment (bhavin-te) [reply tests/io/test_scans.py:2069]: In positive testcase we followed structure of existing testcase (`test_scan_create_scan_document_credentials_pass`) written just above this.

File: tests/io/test_scans.py (hunk 2 above)
Comment (bhavin-te) [reply tests/io/test_scans.py:2088]: In positive testcase we followed structure of existing testcase (`test_scan_create_scan_document_credentials_pass`)

**Pull request merged**